app/javascript/controllers/leaflet_viewer_controller.js

```diff
@@ -151,33 +155,47 @@ export default class LeafletViewerController extends Controller {
     if (this.boundsOverlay) this.overlay.removeLayer(this.boundsOverlay);
   }
 
+  addInspection() {
+    if (this.protocolValue == "Wms") return wmsInspection(this.map, this.urlValue, this.layerIdValue);
```

Comment:
This looks like a good place to use `switch`

Reply:
I had that initially but it's actually more lines because each `case` has to be its own line. And, if anyone forgot to `return` or `break`, you get fun fall-through behavior. I kind of consider JS `switch` to be harmful, tbh

Reply:
What this should ultimately be (imo) is just a lookup in a map that maps the protocol names to callables. Which can happen at some point soon